### PR TITLE
Fix column parameter handling in `read_orc`

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
When `columns=[]` for `read_orc`, pandas actually only drops the column and preserves the `index` while reading an orc file. Fixing the `cudf` behavior to match the same.

On `pandas_2.0_feature_branch`:
```
= 198 failed, 101241 passed, 2091 skipped, 954 xfailed, 312 xpassed in 1098.81s (0:18:18) =
```

This PR:
```
= 185 failed, 101254 passed, 2091 skipped, 954 xfailed, 312 xpassed in 1088.47s (0:18:08) =
```
## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/_lib/orc.pyx [hunk 1]
```diff
@@ -129,14 +129,20 @@ cpdef read_orc(object filepaths_or_buffers,
 
     data, index = data_from_unique_ptr(
         move(c_result.tbl),
-        col_names if columns is None else names,
+        col_names if columns is None
+        or (isinstance(columns, list) and len(columns) == 0) else names,
         actual_index_names
     )
```
Comment (isVoid) [inline python/cudf/cudf/_lib/orc.pyx:135]: Would the following be clearer in logic?

Use `if columns is not None and (isinstance(columns, list) and len(columns) == 0)` as the main branching condition:

```python
# shortcut to return
if columns is not None and (isinstance(columns, list) and len(columns) == 0):
        index = cudf.RangeIndex(0, len(data[list(data.keys())[0]]))
        data = {}
        return index, data
        
data, index = data_from_unique_ptr(...) # unchanged
...
```

Comment (galipremsagar) [reply python/cudf/cudf/_lib/orc.pyx:135]: We could do that if we knew how many rows are in the entire orc file. Unfortunately, we don't have that data from orc file metadata, that's why we need to read the data and then fetch the number of rows with `len(data[list(data.keys())[0]])`

Comment (isVoid) [reply python/cudf/cudf/_lib/orc.pyx:135]: Ok. The num row information should exist in `c_result`. But there's no easy way extracting that information without `data_from_unique_ptr`. I'm a bit rusted on cython+owning pointer tricks in cuDF, but would the following code work?

```python
if columns is not None and (isinstance(columns, list) and len(columns) == 0):
        cdef size_type nrow = c_result.tbl[0].view().nrows()
        return cudf.RangeIndex(nrow), {}
```

This is also non-blocking, so feel free to move on.

Comment (galipremsagar) [reply python/cudf/cudf/_lib/orc.pyx:135]: Getting the following error:
```
              nrows = c_result.tbl[0].view().num_rows()
                                        ^
      ------------------------------------------------------------
      
      /nvme/0/pgali/cudf/python/cudf/cudf/_lib/orc.pyx:134:34: Indexing 'unique_ptr[table]' not supported for index type 'long'
      ninja: build stopped: subcommand failed.
```

Comment (isVoid) [reply python/cudf/cudf/_lib/orc.pyx:135]: Ah, so `c_result.tbl` is a `unique_ptr<table>` type. with `.get` method you can retrieve the underlying raw pointer.

```python
    cdef size_type nrows = c_result.tbl.get()[0].view().num_rows()
    if columns is not None and (isinstance(columns, list) and len(columns) == 0):
        return cudf.RangeIndex(nrows), {}
```

Comment (galipremsagar) [reply python/cudf/cudf/_lib/orc.pyx]: Incorporated this suggestion 👍 Thanks @isVoid for helping with this issue.

**Review by mroeschke — APPROVED** [review]

**Pull request merged**